src/entry.sh

```diff
@@ -9,5 +9,5 @@ echo -e "${GREEN}Systemd init system enabled."
 # We need this sleep to ensure this doesn't occur, else
 # logging to the console will not work.
 sleep infinity &
-env > /etc/docker.env
+env | sed "s/=\(.*\)$/='\1'/g" > /etc/docker.env
```

Comment:
This will singlequote everything regardless, so:
```
HELLO='there'
```
will become:
```
HELLO=''there''
```
which is most definitely not what we want.
Additionally, integers will implicitly be converted from `123` to `'123'` which again might cause issues with whatever's using the envvars (if it's another bash script, for example).

I think a better approach is to check for a backslash and only singlequote contain the value *if* it's not already quoted.

Reply:
@hedss Thanks dude! Your concern is fair, as this change could have side-effects in various places. 
In specific though, both definitions of `HELLO='there'` and `HELLO=''there''` will result in an envvar `HELLO=there`
The same applies for envvars `NUM_VALUE='123'` and `NUM_VALUE=123`, that will become `NUM_VALUE=123`. 
envvars are strings, right?
We should probably single quote every value, as we might have some unexpected side effects for other cases (except backslashes) like `$0` for example. 
We probably also need a look by @Page- before merging this.

Reply:
@mikesimos Well, bash lets variables be strings or integers, though implicit assignment occurs on certain keywords (`let` for example). 
But I don't really think we want to be quoting numbers if they're not already specifically typed as a string, that's my worry. :-/

Reply:
@hedss fair point, but it seems that quoting them, won't make a difference in current context, as contents of /etc/docker.env are character strings [more info](https://linux.die.net/abs-guide/untyped.html).  I mostly insist here, for the other special cases (like the `$0` example above).

Reply:
Going back to [this](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#EnvironmentFile), only envvars containing `\` should be affected. I think we can move on with quoting just those envvars.

Reply:
Thanks @hedss!
Thanks @wrboyce, this looks better :+1: